source/detail/serialization/xlsx_consumer.cpp

```diff
@@ -183,15 +184,23 @@ xlnt::detail::Cell parse_cell(xlnt::row_t row_arg, xml::parser *parser, std::uno
         }
         else if (string_equal(attr.first.name(), "s"))
         {
-            c.style_index = static_cast<int>(strtol(attr.second.value.c_str(), nullptr, 10));
+            int index = -1;
+            if (xlnt::detail::parse(attr.second.value, index))
+            {
+                c.style_index = index;
+            }
         }
         else if (string_equal(attr.first.name(), "ph"))
         {
             c.is_phonetic = is_true(attr.second.value);
         }
         else if (string_equal(attr.first.name(), "cm"))
```

Comment:
FYI, this attribute is not covered by a unit test: https://coveralls.io/jobs/156064273/source_files/35998489148. As we are changing the implementation, it may be a good time to add an extra unit test.